Unsatisfied dependencies with Tomcat and Weld

Hi,

we are running Tomcat 8 with Weld 2.2.8. Dependency Injection works and our code runs fine.

```
@ApplicationScoped
public class SqlSessionFactoryProducer {
    
    @Produces @ApplicationScoped
    public SqlSessionFactory produce() {
        try {
            String resource = "de/grasmueck/data/Configuration.xml";
            InputStream inputStream = Resources.getResourceAsStream(resource);
            SqlSessionFactory sqlSessionFactory = new SqlSessionFactoryBuilder().build(inputStream);
            return sqlSessionFactory;
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}
```

All our DAO Classes share the same basic structure.

```
@RequestScoped
public class MyExamplePojoDAO {
    
    @Inject
    private SqlSessionFactory sqlSessionFactory;

    public List<MyExamplePojo> select() {
        try (SqlSession session = sqlSessionFactory.openSession()) {
            MyExamplePojoMapper mapper = session.getMapper(MyExamplePojoMapper.class);
            return mapper.select();
        }
    }
}
```

I am trying to get rid of the boilerplate code with Mybatis-CDI and inject with `@Mapper`. During deployment I end up with the following exception.

```
Caused by: org.jboss.weld.exceptions.DeploymentException: WELD-001408: Unsatisfied dependencies for type UserTransaction with qualifiers @Default
  at injection point [BackedAnnotatedField] @Inject private transient org.mybatis.cdi.JtaTransactionInterceptor.userTransaction
  at org.mybatis.cdi.JtaTransactionInterceptor.userTransaction(JtaTransactionInterceptor.java:0)
```

Makes sense because with don't work with JTA Transaction at all. Any chance to make this work? We don't want to work with a full blown application server right now.

**Edit** 
I forgot our Configuration.xml. It is rather simple. We have a single environment with a JDBC Transaction Manager and a POOLED datasource.

--- a/pom.xml
+++ b/pom.xml
@@ -21,7 +21,7 @@
   <parent>
     <groupId>org.mybatis</groupId>
     <artifactId>mybatis-parent</artifactId>
-    <version>22-SNAPSHOT</version>
+    <version>24-SNAPSHOT</version>
   </parent>
 
   <artifactId>mybatis-cdi</artifactId>

--- a/pom.xml
+++ b/pom.xml
@@ -102,13 +102,13 @@
     <dependency>
       <groupId>org.slf4j</groupId>
       <artifactId>slf4j-api</artifactId>
-      <version>1.7.10</version>
+      <version>1.7.12</version>
       <scope>test</scope>
     </dependency>
     <dependency>
       <groupId>org.slf4j</groupId>
       <artifactId>slf4j-simple</artifactId>
-      <version>1.7.10</version>
+      <version>1.7.12</version>
       <scope>test</scope>
     </dependency>
   </dependencies>

--- a/src/main/java/org/mybatis/cdi/JtaTransactionInterceptor.java
+++ b/src/main/java/org/mybatis/cdi/JtaTransactionInterceptor.java
@@ -15,6 +15,7 @@
  */
 package org.mybatis.cdi;
 
+import javax.enterprise.inject.Instance;
 import javax.inject.Inject;
 import javax.interceptor.Interceptor;
 import javax.transaction.HeuristicMixedException;
@@ -38,29 +39,29 @@ public class JtaTransactionInterceptor extends LocalTransactionInterceptor {
   private static final long serialVersionUID = 1L;
 
   @Inject
-  private transient UserTransaction userTransaction;
+  private transient Instance<UserTransaction> userTransaction;
 
   @Override
   protected boolean isTransactionActive() throws SystemException {
-    return userTransaction.getStatus() != Status.STATUS_NO_TRANSACTION;
+    return userTransaction.get().getStatus() != Status.STATUS_NO_TRANSACTION;
   }
 
   @Override
   protected void beginJta() throws NotSupportedException, SystemException {
-    userTransaction.begin();
+    userTransaction.get().begin();
   }
 
   @Override
   protected void endJta(boolean isExternaTransaction, boolean needsRollback) throws SystemException, RollbackException, HeuristicMixedException, HeuristicRollbackException {
     if (isExternaTransaction) {
       if (needsRollback) {
-        userTransaction.setRollbackOnly();
+        userTransaction.get().setRollbackOnly();
       }
     } else {
       if (needsRollback) {
-        userTransaction.rollback();
+        userTransaction.get().rollback();
       } else {
-        userTransaction.commit();
+        userTransaction.get().commit();
       }
     }
   }

